src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -105,41 +174,40 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._pool.close()
             self._pool.join()
 
-    def _new_process(self):
-        input_queue = Queue()
-        output_queue = Queue()
-        current_log_context = LogContext.get_current()
-        process = Process(
-            target=_process_wrapper,
-            args=(
-                self._executor_creation_func,
-                input_queue,
-                output_queue,
-                current_log_context.get_initializer() if current_log_context else None,
-                OperationContext.get_instance().get_context_dict(),
-            ),
-        )
-        process.start()
-        return process, input_queue, output_queue
+    def init_process(self):
+        return LineProcess(self._executor_creation_func)
 
-    def end_process(self, process):
-        if process.is_alive():
-            process.kill()
+    def finish_line_processing(self, line_number, result_list, total_count):
+        self._processing_idx.pop(line_number)
+        log_progress(
+            logger=bulk_logger,
+            count=len(result_list),
+            total_count=total_count,
+            formatter="Finished {count} / {total_count} lines.",
+        )
 
     def _timeout_process_wrapper(self, task_queue: Queue, idx: int, timeout_time, result_list):
-        process, input_queue, output_queue = self._new_process()
+        new_process = self.init_process().start(task_queue, idx)
```

Comment:
The logic could be refined like this:
```
process_manger.start_new_process()
if process_manager.is_ready():
    return

args = task_queue.get()
process_manager.put(args)
self._processing_idx[line_number] = process_manager.format_current_process()
```

When timeouted:
```
process_manager.restart()
```

Reply:
fixed